### ISSUE TITLE
torsocks qtox results in segfault

### ISSUE THREAD

**Issue by adrelanos** [qTox/qTox#5619]
##### Brief Description

* OS: debian buster
* qTox version v1.16.3-2-debian.
* Commit hash: build without git
* toxcore version: 0.2.9
* Qt version: 5.11.3

```
dpkg -l | grep qtox
ii  qtox                                          1.16.3-2
```

* Reproducible: Always

##### Steps to reproduce

Debian buster
sudo apt-get install torsocks qtox

    torsocks qtox

##### Observed Behavior

```
[12:50:33.070 UTC] persistence/settings.cpp:129 : Debug: No settings file found, using defaults
[12:50:33.070 UTC] persistence/settings.cpp:133 : Debug: "Loading settings from :/conf/qtox.ini"
[12:50:33.077 UTC] ipc.cpp:61 : Debug: Our global IPC ID is  15713412508552978184
[12:50:33.078 UTC] main.cpp:251 : Debug: commit:  build without git
[12:50:40.685 UTC] persistence/settings.cpp:2411 : Debug: Creating new profile settings in  "/home/user/.config/tox//aaaaaa.ini"
[12:50:40.712 UTC] persistence/settings.cpp:462 : Debug: "Saving global settings at /home/user/.config/tox/qtox.ini"
[12:50:40.755 UTC] persistence/settings.cpp:335 : Debug: Loading personal settings from "/home/user/.config/tox/aaaaaa.ini"
[12:50:40.756 UTC] nexus.cpp:87 : Debug: Starting up
[12:50:40.970 UTC] core/core.cpp:197 : Debug: Creating a new profile
[12:50:40.970 UTC] core/toxoptions.cpp:63 : Debug: Core starting with IPv6 enabled
[12:50:40.971 UTC] core/toxlogger.cpp:57 : Warning: "ERROR:network.c:new_networking_ex:764: Failed to get a socket?! 1, Operation not permitted"
[12:50:40.971 UTC] core/toxlogger.cpp:57 : Warning: "ERROR:network.c:new_networking_ex:764: Failed to get a socket?! 1, Operation not permitted"
[12:50:40.971 UTC] core/core.cpp:152 : Critical: can't to bind the port
[12:50:41.843 UTC] widget/systemtrayicon.cpp:83 : Debug: Using the Qt backend
Segmentation fault
```

journal:

```
Apr 14 12:48:02 host kernel: qTox Core[4416]: segfault at 0 ip 00007f3acac88cd0 sp 00007f3aa4b69ff0 error 4 in libtoxcore.so.2.9.0[7f3acac61000+31000]
```
##### Expected Behavior

qtox functional.

**Comment by sudden6** [comment]
Can you try if this also happens with qTox build on the master branch?

**Comment by Nurmagoz** [comment]
> Can you try if this also happens with qTox build on the master branch?

same thing: 

```
user@host:~/qTox/qTox$ torsocks qtox
[17:12:58.316 UTC] persistence/settings.cpp:129 : Debug: No settings file found, using defaults
[17:12:58.316 UTC] persistence/settings.cpp:133 : Debug: "Loading settings from :/conf/qtox.ini"
[17:12:58.326 UTC] ipc.cpp:61 : Debug: Our global IPC ID is  11964195608317591975
[17:12:58.326 UTC] main.cpp:251 : Debug: commit:  build without git
[17:13:27.236 UTC] persistence/settings.cpp:2411 : Debug: Creating new profile settings in  "/home/user/.config/tox//user.ini"
[17:13:27.238 UTC] persistence/settings.cpp:462 : Debug: "Saving global settings at /home/user/.config/tox/qtox.ini"
[17:13:27.259 UTC] persistence/settings.cpp:335 : Debug: Loading personal settings from "/home/user/.config/tox/user.ini"
[17:13:27.260 UTC] nexus.cpp:87 : Debug: Starting up
[17:13:28.432 UTC] core/core.cpp:197 : Debug: Creating a new profile
[17:13:28.432 UTC] core/toxoptions.cpp:63 : Debug: Core starting with IPv6 enabled
[17:13:28.432 UTC] core/toxlogger.cpp:57 : Warning: "ERROR:network.c:new_networking_ex:764: Failed to get a socket?! 1, Operation not permitted"
[17:13:28.432 UTC] core/toxlogger.cpp:57 : Warning: "ERROR:network.c:new_networking_ex:764: Failed to get a socket?! 1, Operation not permitted"
[17:13:28.432 UTC] core/core.cpp:152 : Critical: can't to bind the port
[17:13:29.032 UTC] widget/systemtrayicon.cpp:62 : Debug: Using GTK backend
Segmentation fault
user@host:~/qTox/qTox$ 
```

**Comment by anthonybilinski** [comment]
By default, toxcore (and qTox) operate in UDP mode to participate in the DHT. When qTox is first started under torsocks on a new profile, while initializing toxcore, toxcore fails to open a UDP port (blocked by torsocks) and fails to construct, which causes qTox to segfault due to bad error handling on our side.

If I first open the profile without torsocks (leaking IP over UDP), change to TCP mode (by unchecking "enable UDP" under settings -> advanced), then restart under torsocks, toxcore constructs ok, doesn't attempt to open a UDP socket, and qTox doesn't crash. qTox then connects to the network successfully through a Tox TCP relay and through Tor, and works correctly.

Given that, I think this issue is effectively a duplicate of https://github.com/qTox/qTox/issues/2070 - (although for this the proxy settings don't need to be set before connecting, just the enable of TCP mode thanks to torsocks).

I'll close this issue as duplicate and track progress (and prioritize work) on that issue to address both issues.
